src/main/daml/Daml/Finance/Interface/Instrument/Swap/Asset/Types.daml

```diff
@@ -44,3 +42,16 @@ data Asset = Asset
       -- ^ (Market) time of the last recorded lifecycle event. If no event has occurred yet, the
       --   time of creation should be used.
   deriving (Eq, Show)
+
+-- | Describes an underlying of a product.
+data Underlying = Underlying
+  with
+    referenceAsset : InstrumentKey
+      -- ^ The underlying's key.
+    referenceAssetId : Text
+      -- ^ The reference asset ID. This is used to retrieve observations for this underlying.
+    weight : Decimal
+      -- ^ The basket weight of the underlying.
+    initialPrice : Decimal
+      -- ^ The initial price of the underlying.
```

Comment:
Minor point, but leaving the initialPrice in here does not allow you to do things like initial fixing asianing. Nothing we need to worry about now though